### PR TITLE
TMEDIA-198 - Top table list extract options

### PR DESCRIPTION
**Pull request by matthewroach**
## Description
Update top table list to make use of shared components.

## Jira Ticket
- [TMEDIA-198](https://arcpublishing.atlassian.net/browse/TMEDIA-198)

## Test Steps

1. Checkout this branch `git checkout TMEDIA-198-top-table-list-extract-options`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/top-table-list-block,@wpmedia/shared-styles`
3. Verify top table list works as before
    * Example page - http://localhost/pf/overline-top-table/?_website=the-gazette
    * http://localhost/pf/top-table-list-configs/?_website=the-gazette

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [X] Confirmed all the test steps a reviewer will follow above are working.
- [X] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [X] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [X] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [X] Confirmed this PR has unit test files
  - [X] Ran `npm run test`, made sure all tests are passing
- [X] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/top-table-list-block/features/top-table-list/_children/horizontal-overline-image-story-item.test.jsx [hunk 1]
```diff
@@ -143,21 +148,21 @@ describe('horizontal overline image story item', () => {
     };
 
     const { default: HorizontalOverlineImageStoryItem } = require('./horizontal-overline-image-story-item');
-    const wrapper = shallow(<HorizontalOverlineImageStoryItem {...testProps} />);
+    const wrapper = mount(<HorizontalOverlineImageStoryItem {...testProps} />);
```
Comment (JackHowa) [inline blocks/top-table-list-block/features/top-table-list/_children/horizontal-overline-image-story-item.test.jsx:151]: nice, always happier with mount

File: blocks/top-table-list-block/features/top-table-list/_children/medium-list-item.jsx [hunk 2]
```diff
@@ -1,169 +1,66 @@
...
-          className="description-text"
-        >
-          {descriptionText}
-        </DescriptionText>
-      );
-    }
-    return null;
-  };
-
-  const byLineTmpl = () => {
```
Comment (JackHowa) [inline blocks/top-table-list-block/features/top-table-list/_children/medium-list-item.jsx:74]: happy to get this out

File: blocks/top-table-list-block/features/top-table-list/_children/small-list-item.jsx [hunk 3]
```diff
@@ -108,11 +32,27 @@ const SmallListItem = (props) => {
...
+            content={element}
+            showPromoLabel
+            promoSize="SM"
+            imageRatio={imageRatioSM}
+            editable={false}
+          />
+        ) : null }
       </div>
-      {hrBorderTmpl()}
+      <hr className={!showBottomBorder ? 'hr-borderless' : ''} />
```
Comment (JackHowa) [inline blocks/top-table-list-block/features/top-table-list/_children/small-list-item.jsx:55]: nice

File: blocks/top-table-list-block/features/top-table-list/_children/vertical-overline-image-story-item.test.jsx [hunk 4]
```diff
@@ -23,23 +23,43 @@ const config = {
   showImageSM: true,
 };
 
+const imageURL = 'pic';
+const itemTitle = 'title';
+const descriptionText = 'description';
+const websiteURL = 'https://arcxp.com';
```
Comment (JackHowa) [inline blocks/top-table-list-block/features/top-table-list/_children/vertical-overline-image-story-item.test.jsx:29]: v on brand

File: blocks/top-table-list-block/features/top-table-list/default.scss [hunk 5]
```diff
@@ -43,56 +43,27 @@
...
+    column-gap: calculateRem(24px);
 
     &.horizontal.reverse {
       flex-direction: row-reverse;
 
       > div {
         padding-left: 0;
         padding-right: 0;
-
-        &:nth-child(odd) {
```
Comment (JackHowa) [inline blocks/top-table-list-block/features/top-table-list/default.scss:64]: did we add this in somewhere? or has the style changed?

**Review by JackHowa — COMMENTED** [review]
Hard to look at holistically with all of the changes. Will pull it down and run it. is the styles with the padding left and odd numbering a potential regression or new style? blocks/top-table-list-block/features/top-table-list/default.scss

File: blocks/top-table-list-block/features/top-table-list/default.scss (hunk 5 above)
Comment (JackHowa) [reply blocks/top-table-list-block/features/top-table-list/default.scss:64]: trying to see the difference here. did you just fix the grid?

**Review by JackHowa — APPROVED** [review]
Good work. looks valid testing locally. Would be curious of the bundle change

File: blocks/top-table-list-block/features/top-table-list/default.scss (hunk 5 above)
Comment (matthewroach) [reply blocks/top-table-list-block/features/top-table-list/default.scss:64]: Applied row class to parent and then used `column-gap`

Comment (JackHowa) [reply blocks/top-table-list-block/features/top-table-list/default.scss:64]: gotcha makes sense

**Pull request merged**